### ISSUE TITLE
404 consent-status response when updating library from version 7.2.8 to 7.3.0

### ISSUE THREAD

**Issue by kasim-canol-sz** [SourcePointUSA/android-cmp-app#706]
Hello,

after updating the library from `7.2.8` to `7.3.0`, the SourcePoint SDK tries to check the consent-status, but the request returns a 404 error:

 ```
com.sourcepoint.cmplibrary.exception.RequestFailedException: {"err":"{\"msg\":\"http://internal-prod-cmp-private-308276059.eu-central-1.elb.amazonaws.com/tcfv2/consent/v3/21749/consent-status?hasLocalData=true&withSiteActions=true&hasCsp=false&dateCreated=2023-10-05T11%3A57%3A56.766Z&clientIP=XXX.XXX.XXX.XXX&consentUUID=5daf8abd-0eb3-4084-8d92-e7fc4a29ab58_24 returned a 404\",\"status\":404}"}
           at com.sourcepoint.cmplibrary.data.network.util.ResponseManagerImpl.parseConsentStatusResp(ResponseManagerImpl.kt:98)
           at com.sourcepoint.cmplibrary.data.network.NetworkClientImpl$getConsentStatus$1.invoke(NetworkClientImpl.kt:142)
           at com.sourcepoint.cmplibrary.data.network.NetworkClientImpl$getConsentStatus$1.invoke(NetworkClientImpl.kt:125)
           at com.sourcepoint.cmplibrary.util.FunctionalUtilsKt.check(FunctionalUtils.kt:33)
           at com.sourcepoint.cmplibrary.data.network.NetworkClientImpl.getConsentStatus(NetworkClientImpl.kt:125)
           at com.sourcepoint.cmplibrary.data.ServiceImpl$getMessages$1.invoke(ServiceImpl.kt:199)
           at com.sourcepoint.cmplibrary.data.ServiceImpl$getMessages$1.invoke(ServiceImpl.kt:156)
           at com.sourcepoint.cmplibrary.core.ExecutorManagerImpl$executeOnWorkerThread$1.invoke$lambda-0(ExecutorManager.kt:37)
           at com.sourcepoint.cmplibrary.core.ExecutorManagerImpl$executeOnWorkerThread$1.$r8$lambda$YcU9LB-LQlcZOnRwBNhL6k0AYg0(Unknown Source:0)
           at com.sourcepoint.cmplibrary.core.ExecutorManagerImpl$executeOnWorkerThread$1$$ExternalSyntheticLambda0.run(Unknown Source:2)
           at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
           at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
           at java.lang.Thread.run(Thread.java:1012)
```

As far as I understand, the re-check of the consent-status is triggered by following shared_pref value, which has been introduced in `7.3.0`:

`const val KEY_LOCAL_DATA_VERSION = "sp.key.localDataVersion"`

**_IMPORTANT NOTE:_**  This issue can only be reproduced when the app is upgraded without being uninstalled:
1. Start app with library version `7.2.8` and accept consent
2. Bump version to `7.3.0` and re-install app without uninstall

**Comment by carmelo-iriti** [comment]
Hi @kasim-canol-sz , 
thanks for the message, we already fixed the issue and we are going to release the fix soon

**Comment by carmelo-iriti** [comment]
@kasim-canol-sz we just released the v7.4.0 which contains the fix for your issue, in a couple of hours will be available on maven central

**Comment by kasim-canol-sz** [comment]
@carmelo-iriti Thank you, after updating to v7.4.0 it's working fine now.
